packages/flutter_tools/lib/src/android/deferred_components.dart

```diff
@@ -0,0 +1,674 @@
+    if (goldenFile.existsSync()) {
+      goldenFile.deleteSync();
+    }
+    goldenFile.createSync(recursive: true);
+
+    goldenFile.writeAsStringSync(
+  '''
+# ====================================================================
+# The contents of this file are automatically generated and it is not
+# recommended to modify this file manually.
+# ====================================================================
+#
+# This list of loading units are a snapshot of the loading units generated
+# by gen_snapshot during the assembly of split AOT apps. During the build
+# command, the previously generated loading units in this file are compared
+# against the newly generated loading units to check for any new or removed
+# loading units. In the case where loading units do not match, the build will
+# fail and ask the developer to verify that the deferred components configuration
+# in `pubspec.yaml` is correct. Developers should make any necessary changes
+# to integrate new and changed loading units or remove no longer existing
+# loading units from the configuration. The build command should then be
+# re-run to continue the build process.
+loading-units:
+''', flush: false, mode: FileMode.append);
+    Set<int> usedIds = <int>{};
+    for (LoadingUnit unit in generatedLoadingUnits) {
+      if (usedIds.contains(unit.id)) {
+        continue;
+      }
+      goldenFile.writeAsStringSync('  - id: ${unit.id}\n', flush: false, mode: FileMode.append);
+      goldenFile.writeAsStringSync('    libraries:\n', flush: false, mode: FileMode.append);
+      for (String lib in unit.libraries) {
+        goldenFile.writeAsStringSync('      - $lib\n', flush: false, mode: FileMode.append);
+      }
+      usedIds.add(unit.id);
+    }
+  }
+
+  void _handleResults(List<String> generatedFiles, List<String> modifiedFiles, Map<String, dynamic> goldenComparisonResults, bool exitOnFail) {
+    bool fileChangesNeeded = generatedFiles.isNotEmpty || modifiedFiles.isNotEmpty || (goldenComparisonResults != null && !goldenComparisonResults['match']);
+    if (fileChangesNeeded) {
+      globals.printStatus('''
+===========================================================================
+            $title
+===========================================================================''');
+      if (generatedFiles.isNotEmpty) {
+        globals.printStatus('  Newly generated android files:');
+        for (String filePath in generatedFiles) {
+          globals.printStatus('    - $filePath');
+        }
+      }
+      if (modifiedFiles.isNotEmpty) {
+        globals.printStatus('\n  Modified android files:');
+        for (String filePath in modifiedFiles) {
+          globals.printStatus('    - $filePath');
+        }
+      }
+      if (generatedFiles.isNotEmpty && modifiedFiles.isNotEmpty) {
+        globals.printStatus('''
+
+The above files have been placed into `build/$_kDeferredComponentsTempDirectory`
```

Comment:
This means if their file is 1000 lines and we're making it 1010 lines, they have to diff it themselves. It's ok but if we can make a .patch file, even better.